### PR TITLE
Feat/issue 2918

### LINKED ISSUE #2918: [Feature] Wish marker's `coordinates` or `positions` can support big callback, so i can draw a regression line
### What problem does this feature solve?

If `coordinates` or `positions` can support callback, i can use markLine to draw regression line for scatter chart,

image

### What does the proposed API look like?

```ts
    markLine: {
      positions: (data, series) => {
        // do something
        return points;
      },
      
       coordinates: (data, series) => {
        // do something
        return points;
      },
      
    }
```

### PR DESCRIPTION
**Pull request by kkxxkk2019**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #2918 

### 🐞 Bugserver case id

668d2ee49d33c800b77fcea3
668d2f429d33c800b77fcea4

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/marker/interface.ts [hunk 1]
```diff
@@ -129,8 +134,11 @@ export type ICoordinateOption = {
 export type IMarkerPositionsSpec = {
   /**
    * 画布坐标
+   * `positions` 自 1.12.0 版本开始支持回调函数
    */
-  positions: MarkerPositionPoint[];
+  positions:
```
Comment (xile611) [inline packages/vchart/src/component/marker/interface.ts:139]: options文档需要更新一下

File: packages/vchart/src/component/marker/mark-area/interface/spec.ts [hunk 2]
```diff
@@ -127,8 +129,9 @@ export interface IMarkAreaAngleRadiusSpec extends IMarkerCrossSeriesSpec {
 export type IMarkAreaCoordinateSpec = {
   /**
    * 指定数据点的参考线。基于指定数据点进行参考线的绘制，可以对数据点进行数据处理
+   * `coordinates` 自 1.12.0 版本开始支持回调函数
    */
-  coordinates: IDataPointSpec[];
+  coordinates: IDataPointSpec[] | ((seriesData: Datum[], relativeSeries: IMarkerSupportSeries) => IDataPointSpec[]);
```
Comment (xile611) [inline packages/vchart/src/component/marker/mark-area/interface/spec.ts:134]: 更新文档

**Review by xile611 — APPROVED** [review]

File: packages/vchart/src/component/marker/interface.ts (hunk 1 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/interface.ts:139]: 好的，我晚些时候补充文档和 demo

File: packages/vchart/src/component/marker/mark-area/interface/spec.ts (hunk 2 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-area/interface/spec.ts:134]: 好的

File: packages/vchart/src/component/marker/interface.ts (hunk 1 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/interface.ts:139]: done

File: packages/vchart/src/component/marker/mark-area/interface/spec.ts (hunk 2 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-area/interface/spec.ts:134]: done

**Review by xile611 — APPROVED** [review]

**Pull request merged**